File: dotnet/cafdemalihapa/Directories/Repository/Dotnet/Cafdemalihapa/Scripts/Alone/AloneDirectory.cs
using cafdemalihapa.Names;

namespace cafdemalihapa.Directories.Repository.Dotnet.Cafdemalihapa.Scripts.Alone
{
    public class AloneDirectory
    (
        SecondaryApplication secondaryApplication,
        ScriptsDirectory scriptsDirectory
    )
    {
        public string GetName()
        {
            var primaryDirectory =
                Path.Combine(
                    scriptsDirectory.GetPath(),
                    "primary"
                );

            var templateSourceDirectory =
                Path.Combine(
                    primaryDirectory,
                    "environment-variables-template-files"
                );

            return templateSourceDirectory;
        }

        public string AloneOrDouble()
        {
            var x = secondaryApplication.GetName();
            return x.Contains("couldn") ? "alone" : "double";
        }

        public bool IsAlone()
        {
            return AloneOrDouble() == "alone";
        }
    }
}

File: dotnet/cafdemvers/zzq-cafdemwimu/cafdemwimu.console/Directories/Repository/Dotnet/Cafdemalihapa/Scripts/Alone/AloneDirectory.fs
namespace cafdemwimu.console.Directories.Repository.Dotnet.Cafdemalihapa.Scripts.Alone

open System.IO
open cafdemwimu.console.Names
open cafdemwimu.console.Directories.Repository.Dotnet.Cafdemalihapa.Scripts

type AloneDirectory(secondaryApplication: SecondaryApplication, scriptsDirectory: ScriptsDirectory) =
    member _.GetName() =
        let primaryDirectory =
            Path.Combine(scriptsDirectory.GetPath(), "primary")

        let templateSourceDirectory =
            Path.Combine(primaryDirectory, "environment-variables-template-files")

        templateSourceDirectory

    member this.AloneOrDouble() =
        let x = secondaryApplication.GetName()
        if x.Contains("couldn") then "alone" else "double"

    member this.IsAlone() =
        this.AloneOrDouble() = "alone"
